Topics/Tech_Stacks/Web-Sockets.md

```diff
@@ -0,0 +1,162 @@
+# Building Interactive Web Applications with WebSockets
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Understanding WebSockets](#understanding-websockets)
+3. [WebSockets vs HTTP](#websockets-vs-http)
+4. [Advantages and Disadvantages](#advantages-and-disadvantages)
+5. [Setting Up](#setting-up)
+6. [Real-World Use Cases](#real-world-use-cases)
+7. [Conclusion](#conclusion)
+8. [Further Reading](#further-reading)
+
+## Introduction
+In the evolving landscape of web development, the need for real-time, interactive web applications is more prominent than ever. Users expect dynamic experiences with instant updates—be it in chat applications, live sports scores, or stock market feeds. WebSockets represent a significant advancement in web communication, allowing for real-time, full-duplex communication between the client and server. This technology is especially relevant to the CSC301 curriculum as it introduces students to modern web development practices, preparing them for the challenges of developing interactive, user-centric software solutions.
+
+## Understanding WebSockets
+WebSockets are a powerful protocol embedded within the familiar web technology stack, enabling a significant leap in communication capabilities over the Internet. They provide a way to overcome the limitations of the traditional HTTP request-response model, especially when it comes to real-time, low-latency communication.
+
+Unlike HTTP, where a new connection is established for each request and response, WebSockets establish a single, persistent connection between the client and the server. This connection is initiated by the client sending a special type of HTTP request, asking the server to upgrade the connection to a WebSocket connection.
+
+### How the WebSocket Connection is Established:
```

Comment:
Is this only done for web or it can be done for other software solution such as Mobile as well?

Reply:
Yes they can be used for both.